### DISCUSSION TITLE
thermodynamic integration

### DISCUSSION THREAD

**Discussion by chtchelkatchev** [deepmodeling/deepmd-kit#1585] [Ideas]
Hello. When working with deepmd, it is often necessary to perform thermodynamic integration, e.g., in order to build phase diagrams. There is an unpublished officially package, dpti. But its documentation is incomplete and, moreover, the thermodynamic integration procedure there looks overly complicated compared to standard methods. Lammps has built-in thermodynamic integration routines. Are they compatible with deepmd? If yes, please provide at least one example with thermodynamic integration in the deepmd documentation. Thanks.

**Comment by wanghan-iapcm** [comment]
The thermodynamic integration implemented in dpti is a quite standard technique and has been well established. You are referred to https://iopscience.iop.org/article/10.1088/0953-8984/20/15/153101 for a nice review of the thermodynamic integration. 
In principle, the thermodynamic integration implemented in LAMMPS gives the same result as dpti if your simulation time is long enough. The advantage of dpti is that all MD simulations are equilibrium, so the statistical uncertainty can be easily estimated. The documentation of dpti may not be good, please raise questions on which part of the doc is unclear to you. We are willing to improve.

**Reply by chtchelkatchev** [reply]
Thank you for quick answer. Regarding dpti, it is not clear why you use soft-LJ in hti-stage of dpti (in dpti example), how one should fix LJ parameters (e.g. rcut). Taking a two-component metallic system with very different atom radii, what should I do with LJ parameters? Can dpti deal with nonmetallic systems?

**Reply by wanghan-iapcm** [reply]
We use softcore LJ to prevent overlapping of atoms. The critical parameter is \sigma, which should be set roughly the same as the atom radii. In principle, the free energy is independent with all parameters in softcore LJ, but a best choice of the parameters removes numerical singularity and allows one to use less integration points. dpti can work with non-metallic system, for example https://doi.org/10.1103/PhysRevLett.126.236001

**Reply by chtchelkatchev** [reply]
Thank You for fast reply. Current versions of deepmd-2.x.x require "pair_coeff * *" in input-lammps scripts. Old version of deepmd-1.x.x do not have " * *". Input files that use dpti refer to old deepmd. Can you fix this?

**Reply by wanghan-iapcm** [reply]
@felix5572 Could you please fix the issue in dpti? Thanks!

**Reply by felix5572** [reply]
ok i will fix this issue soon @wanghan-iapcm
